tlatools/src/tlc2/tool/Simulator.java

```diff
@@ -252,9 +281,14 @@ public void simulate() throws Exception {
 		ArrayList<SimulationWorker> workers = new ArrayList<>();
 		Set<Integer> runningWorkers = new HashSet<>();
 		for (int i = 0; i < this.numWorkers; i++) {
-			SimulationWorker worker = new SimulationWorker(i, tool, initStates, this.workerResultQueue,
+			
+			// Give each worker its own Tool and liveness checker object.
+			Tool newTool = new Tool(specDir, specFile, configFile, resolver);
```

Comment:
Note: I changed this to give every worker its own Tool, just to be safe in case Tool is not thread safe.

Reply:
Tool is already being used concurrently by exhaustive search. Thus, shouldn't be a problem to use a single instance.